Rakefile

```diff
@@ -85,11 +91,45 @@ task :analytics do
   end
 end
 
-desc "Dump formulae and analytics data"
+desc "Dump Linux analytics data"
+task :analytics_linux do
+  next unless generate_analytics?("linux")
+
+  setup_analytics
+
+  %w[build-error install cask-install install-on-request os-version
+     core-build-error core-install core-install-on-request].each do |category|
+    case category
+    when "core-build-error"
+      category = "all-core-formulae-json --build-error"
+      category_name = "build-error/linuxbrew-core"
+    when "core-install"
+      category = "all-core-formulae-json --install"
+      category_name = "install/linuxbrew-core"
+    when "core-install-on-request"
+      category = "all-core-formulae-json --install-on-request"
+      category_name = "install-on-request/linuxbrew-core"
+    else
+      category_name = category
+    end
+    FileUtils.mkdir_p "_data/analytics-linux/#{category_name}"
+    %w[30 90 365].each do |days|
+      next if days != "30" && category_name == "build-error/linuxbrew-core"
+
+      sh "brew formula-analytics --linux --days-ago=#{days} --json --#{category} " \
+        "> _data/analytics-linux/#{category_name}/#{days}d.json"
+    end
```

Comment:
Any way of unifying some of the above into a method that can be shared?

Reply:
Yep. I did this in 1cb1da3.